src/Features/SupportWireLoading/BrowserTest.php

```diff
@@ -378,6 +381,64 @@ public function render()
         ->assertSee('Foo')
         ;
     }
+
+    /** @test */
+    function wire_loading_works_when_function_called_by_event()
+    {
+        Livewire::visit(new class extends Component {
+            protected $listeners = ['foo' => 'onFoo'];
+
+            function onFoo() {
+                sleep(5);
+            }
+
+            function render()
+            {
+                return <<<'HTML'
+                <div>
+                    <button @click="$dispatch('foo')" dusk="button">Dispatch "foo"</button>
+
+                    <div wire:loading wire:target="onFoo" dusk="loadingIndicator">
+                        Processing...
+                    </div>
+                </div>
+                HTML;
+            }
+        })
+        ->assertDontSee('Processing...')
+        ->waitForLivewire()->click('@button')
+        ->assertSee('Processing...');
```

Comment:
```php
->click('@button')
->waitForText('Processing...')
->assertSee('Processing...');
```

Reply:
@danie-ramdhani FYI using `->waitForLivewire()` is fine and sometimes better as it resolves when the Livewire request actually finishes. I recently improved it to make it more stable too.